_posts/2021/2021-09-20-compact_blank-remove-empty-strings-from-array-and-hash.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Use compact_blank to remove empty strings from Arrays and Hashes"
+description: "This is one of those “Nice Things” in Rails"
+layout: article
+category: ruby
+image:
+  base: '2021/compact_blank-remove-empty-strings-from-array-and-hash'
+  alt: "Compact Disc"
+  credit: "Jan Huber"
+  source: "https://unsplash.com/photos/oTCRizM-PUI"
+
+---
+
+Active Support is a mixed library of “useful” things that’s included with Rails. This includes [extending the base Ruby classes](https://guides.rubyonrails.org/active_support_core_extensions.html) such as `Hash`, `String` and `Integer`.
+
+Even the most experienced Rails developer is constantly finding helpful methods that are new to them. And there are useful new methods being added even in recent versions.
+
+A task I often find myself doing is cleaning up the values of`Hash`es and `Array`s. Ruby provides the `compact` method to remove `nil` values from an array and (since 2.4) has provided a [similarly named method](https://ruby-doc.org/core-2.4.0/Hash.html#method-i-compact) for Hash.
```

Comment:
*similarly-named (I think)

Reply:
```suggestion
A task I often find myself doing is cleaning up the values of`Hash`es and `Array`s. Ruby provides the `#compact` method to remove `nil` values from an array and (since 2.4) has provided a [the same `#compact` method](https://ruby-doc.org/core-2.4.0/Hash.html#method-i-compact) for Hash.
```